src/tiled/tilepainter.cpp

```diff
@@ -197,12 +199,26 @@ static QRegion fillRegion(const TileLayer *layer, QPoint fillOrigin,
     if (!layer->contains(fillOrigin))
```

Comment:
This check should be disabled for infinite maps (once `contains` is back to its original form).